tests/api/integration/test_client.py

```diff
@@ -498,3 +499,58 @@ def test_multiple_jobs(self):
             self.get_grading_job_queue_position(
                 self.course1, job_id, self.client_header1, 400
             )
+
+
+class StreamEndpointTest(BaseTest):
+    def test_stream(self):
+        num_jobs = 5
+
+        # Upload the jobs
+        self.upload_grading_config(
+            self.course1,
+            "assignment1",
+            self.client_header1,
+            grading_configs.only_student_config,
+            200,
+        )
+
+        job_ids = []
+        for _ in range(num_jobs):
+            grading_run_id = self.start_grading_run(
+                self.course1,
+                "assignment1",
+                self.client_header1,
+                grading_runs.one_student_job,
+                200,
+            )
+
+            # Keep track of the job ids
+            run_state = self.get_grading_run_state(
+                self.course1, grading_run_id, self.client_header1
+            )
+            job_ids.append(list(run_state["student_jobs_state"].keys())[0])
+
+        for ind, job_id in enumerate(job_ids):
+
+            def _create_callback(chunks):
+                def _callback(chunk):
+                    self.assertNotEquals(len(chunks), 0)
+                    self.assertEquals(chunk, chunks.pop())
+
+                return _callback
+
+            chunks = deque()
+            chunks.append(b"event: state\ndata: FINISHED\n\n")
+            chunks.append(b"event: state\ndata: STARTED\n\n")
+            for pos in range(ind):
+                chunks.append(f"event: position\ndata: {pos}\n\n".encode())
```

Comment:
I'm a little confused by this code. Why is `FINISHED` before `STARTED`, before positions?

Reply:
I'm using the `deque` as a stack so those will be popped and asserted in reverse order.

Reply:
So the expected output is positions from large to small, start, and then finish?

Reply:
Yes